kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/KafkaProtocolFilterReconciler.java

```diff
@@ -0,0 +1,205 @@
+import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
+import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
+import io.javaoperatorsdk.operator.processing.event.ResourceID;
+import io.javaoperatorsdk.operator.processing.event.source.EventSource;
+import io.javaoperatorsdk.operator.processing.event.source.informer.InformerEventSource;
+
+import io.kroxylicious.kubernetes.api.common.Condition;
+import io.kroxylicious.kubernetes.api.common.ConditionBuilder;
+import io.kroxylicious.kubernetes.filter.api.v1alpha1.KafkaProtocolFilter;
+import io.kroxylicious.kubernetes.filter.api.v1alpha1.KafkaProtocolFilterBuilder;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+
+/**
+ * Reconciles a {@link KafkaProtocolFilter} by checking whether the {@link Secret}s
+ * and {@link ConfigMap}s refered to in interpolated expressions actually exist, setting a
+ * {@link Condition.Type#ResolvedRefs} {@link Condition} accordingly.
+ */
+public class KafkaProtocolFilterReconciler implements
+        Reconciler<KafkaProtocolFilter> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KafkaProtocolFilterReconciler.class);
+    private final Clock clock;
+    private final SecureConfigInterpolator secureConfigInterpolator;
+
+    KafkaProtocolFilterReconciler(Clock clock, SecureConfigInterpolator secureConfigInterpolator) {
+        this.clock = clock;
+        this.secureConfigInterpolator = secureConfigInterpolator;
+    }
+
+    @Override
+    public List<EventSource<?, KafkaProtocolFilter>> prepareEventSources(EventSourceContext<KafkaProtocolFilter> context) {
+        return List.of(
+                new InformerEventSource<>(foo(context, Secret.class,
+                        interpolationResult -> interpolationResult.volumes().stream()
+                                .flatMap(volume -> Optional.ofNullable(volume.getSecret())
+                                        .map(SecretVolumeSource::getSecretName)
+                                        .stream())),
+                        context),
+                new InformerEventSource<>(foo(context, ConfigMap.class,
+                        interpolationResult -> interpolationResult.volumes().stream()
+                                .flatMap(volume -> Optional.ofNullable(volume.getConfigMap())
+                                        .map(ConfigMapVolumeSource::getName)
+                                        .stream())),
+                        context));
+    }
+
+    private <R extends HasMetadata> InformerEventSourceConfiguration<R> secretExtractingEventSourceConfig(
+                                                                            EventSourceContext<KafkaProtocolFilter> context,
+                                                                            Class<R> secondaryClass,
+                                                                            Function<SecureConfigInterpolator.InterpolationResult, Stream<String>> secretIdExtractor) {
```

Comment:
The enforced line wrapping forced on us here by our authoritarian formatter really makes me want to weep. Because _obviously_ a line that's 211 characters long must be easier to read, right? Right?